awx/main/tests/unit/test_tasks.py

```diff
@@ -309,7 +309,15 @@ class TestExtraVarSanitation(TestJobExecution):
 
     UNSAFE = '{{ lookup(''pipe'',''ls -la'') }}'
 
-    def test_vars_unsafe_by_default(self):
+    @pytest.mark.parametrize('safe_mode,marked_unsafe', [
+        ('always', False),
+        ('template', True),
+        ('never', True)
+    ])
+    def test_vars_unsafe_by_default(self, safe_mode, marked_unsafe):
+        patch = mock.patch('awx.main.tasks.settings.ALLOW_JINJA_IN_EXTRA_VARS', safe_mode)
+        patch.start()
```

Comment:
You need to undo this patch afterwards, right?

Usually I like to use the context manager so this gets undone i.e.,

```python
with mock.patch(...):
   ...
```

Reply:
yeah, that sounds safer to me and very easy to do as a context manager